src/controllers/userController.js

```diff
@@ -1,43 +1,53 @@
-const User = require('../models/user');
+const User = require("../models/user");
 
 // @route GET admin/user
 // @desc Returns all users
 // @access Public
 exports.index = async function (req, res) {
-	const users = await User.find({});
-	res.status(200).json({users});
-}
+  const users = await User.find({});
+  res.status(200).json({ users });
+};
 
 // @route GET api/user/{id}
 // @desc Returns a specific user
 // @access Public
 exports.show = async function (req, res) {
-	try {
-		const id = req.params.id;
-		const user = await User.findById(id);
+  try {
+    const id = req.params.id;
+    const user = await User.findById(id);
 
-		if (!user) return res.status(401).json({message: 'User does not exist'});
+    if (!user) return res.status(401).json({ message: "User does not exist" });
 
-		res.status(200).json({user});
-	} catch (error) {
-		res.status(500).json({message: error.message})
-	}
+    res.status(200).json({ user });
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
 };
 
 // @route PUT api/user/{id}
 // @desc Update user details
 // @access Public
 exports.update = async function (req, res) {
-	try {
-		const update = req.body;
-		const id = req.params.id;
-		const userId = req.user._id;
-
-		// Make sure the passed id is that of the logged in user
-		if (userId.toString() !== id.toString()) return res.status(401).json({message: "Sorry, you don't have the permission to update this data."});
-
-		const user = await User.findByIdAndUpdate(id, {$set: update}, {new: true});
-	} catch (error) {
-		res.status(500).json({message: error.message});
-	}
+  try {
+    const update = req.body;
+    const id = req.params.id;
+    const userId = req.user._id;
+
+    // Make sure the passed id is that of the logged in user
+    if (userId.toString() !== id.toString())
+      return res.status(401).json({ message: "Sorry, you don't have the permission to update this data." });
+
+    const user = await User.findByIdAndUpdate(id, { $set: update }, { new: true });
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+exports.profile = async (req, res) => {
+  try {
+    const profile = await User.findOne({ _id: req.user.id }).populate("bookedTours");
```

Comment:
Yes, this is correct. However, you need to populate the "tour" field in the populated bookedTour object. This is because asides from the information about the booking, the details (e.g title and location) of the tour itself would be needed on that page.

You can check this link for hints


https://medium.com/fredwong-it/mongoose-populate-multiple-levels-7e45be41d16b

Reply:
Thanks I guess I was looking at the wrong articles or didn't understand the documentation lol